diff -Nuar linux-3.18.1-llvm.orig/Makefile linux-3.18.1-llvm/Makefile
--- linux-3.18.1-llvm.orig/Makefile	2014-12-20 23:44:22.346631345 +0100
+++ linux-3.18.1-llvm/Makefile	2014-12-20 23:45:10.603634622 +0100
@@ -410,12 +410,13 @@
 KBUILD_CFLAGS   := -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs \
 		   -fno-strict-aliasing -fno-common \
 		   -Werror-implicit-function-declaration \
+		   $(call cc-option,-no-integrated-as,) \
 		   -Wno-format-security \
 		   -std=gnu89
 
 KBUILD_AFLAGS_KERNEL :=
 KBUILD_CFLAGS_KERNEL :=
-KBUILD_AFLAGS   := -D__ASSEMBLY__
+KBUILD_AFLAGS   := -D__ASSEMBLY__ $(call cc-option,-no-integrated-as,)
 KBUILD_AFLAGS_MODULE  := -DMODULE
 KBUILD_CFLAGS_MODULE  := -DMODULE
 KBUILD_LDFLAGS_MODULE := -T $(srctree)/scripts/module-common.lds
